application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -154,43 +155,51 @@ private boolean isAuthorOnCooldown(long userId) {
             .isPresent();
     }
 
-    private MessageCreateAction createModMessage(String modalMessage, long userID,
-            String reportedMessage, String reportedMessageUrl, TextChannel modMailAuditLog) {
-        MessageEmbed embed = new EmbedBuilder().setTitle("Report")
-            .setDescription(("""
+    private MessageCreateAction createModMessage(String reportReason, User author,
+            String reportedMessage, String reportedMessageUrl, String reportedMessageTimestamp,
+            TextChannel modMailAuditLog) {
+        MessageEmbed reportedMessageEmbed = new EmbedBuilder().setTitle("Report")
+            .setDescription("""
                     Reported Message: **%s**
-                    [Reported Message URL](%s)
-                    Reason: **%s**""").formatted(reportedMessage, reportedMessageUrl, modalMessage))
-            .setAuthor("Author ID: %d".formatted(userID))
+                    [Reported Message URL](%s)""".formatted(reportedMessage, reportedMessageUrl))
+            .setAuthor(author.getName(), null, author.getAvatarUrl())
```

Comment:
i might be wrong, but this reports the wrong user, no?

make sure its the author of the reported message and not the user who send the report (who is supposed to stay anonymous)

Reply:
It seems to be the case, Il need to pass the name and avatar url from the `MessageContext...`